### ISSUE TITLE
Add workspace ID in IPC response to GET_WORKSPACES request.

### ISSUE THREAD

**Issue by Iskustvo** [i3/i3#3891]
## I'm submitting a…
<pre>
[ ] Bug
[x] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
At the moment, the **only** link between WORKSPACE event response (i3 node) and GET_WORKSPACES response is the workspace name.
It feels unnatural(and inefficient) to use a workspace name as ID to link stored data from GET_WORKSPACES response and incoming WORKSPACE events.
Also, all of this relies on uniqueness of workspace names, which might change in the future.

## Desired Behavior
The WORKSPACE event provides the client with i3's node data which already contains ID of the node.
It would be very convenient if GET_WORKSPACES would also provide ID for each workspace node.

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: 4.17.1 (2019-08-30)
</pre>

**Comment by Airblader** [comment]
> It feels unnatural(and inefficient)

I disagree with both of these statements. Workspace names are our primary identifier, after all users interact with workspaces using either the name or the number, but rarely even with the container ID. 

There's also no reason that this would change in the future.

That said, 

> It would be very convenient if GET_WORKSPACES would also provide ID for each workspace node.

This sounds fine to me. Feel free to send a PR.

**Comment by Iskustvo** [comment]
Sorry, I forgot to answer here...
Yes, it absolutely makes sense on i3's side to use strings as IDs because i3 interacts with humans.
It doesn't make sense that 2 programs (like i3 and status bar) use strings as IDs, because they have no use of "human readable format", while it adds the overhead to the implementation.

On the side note, without ID in GET_WORKSPACES, it would again be inconvenient/inefficient to properly react on WORKSPACE(change = rename) event. You would either have to request GET_WORKSPACES again and override the stored data or have previously stored map of ID -> name from GET_TREE.

P.S. I have tested my PR for this case and the pointer sent as ID stays the same after renaming.
Therefore, it is safe to use it to find the renamed workspace.
(You probably knew this, but I had to check...)
